deebot_client/commands/xml/charge_state.py

```diff
@@ -0,0 +1,57 @@
+"""Error commands."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from deebot_client.events import StateEvent
+from deebot_client.message import HandlingResult
+from deebot_client.models import State
+
+from .common import XmlCommandWithMessageHandling
+
+if TYPE_CHECKING:
+    from xml.etree.ElementTree import Element
+
+    from deebot_client.event_bus import EventBus
+
+
+class GetChargeState(XmlCommandWithMessageHandling):
+    """Get pos command."""
+
+    name = "GetChargeState"
+
+
+    @classmethod
+    def _handle_xml(cls, event_bus: EventBus, xml: Element) -> HandlingResult:
+        """Handle xml message and notify the correct event subscribers.
+
+        :return: A message response
+        """
+
+        # examples:
+        # <ctl ret='ok'><charge type='SlotCharging' g='0'/></ctl>
+        # <ctl ret='ok'><charge type='Idle' g='0'/></ctl>
+        # <ctl ret='ok'><charge type='Going' g='0'/></ctl>
+
+        status: State | None = None
+
+        ret = str(ret) if (ret := xml.attrib["ret"]) else ""
+        if ret != "ok":
+            return HandlingResult.analyse()
+
+        if charge := xml.find("charge"):
+            type = charge.attrib["type"]
+            if type == "SlotCharging" or type == "slot_charging" or type == "WireCharging":
+                status = State.DOCKED
+            elif type == "Idle" or type == "idle":
+                status = State.IDLE
+            elif type == "Going" or type == "going":
+                status = State.RETURNING
+            else:
+                status = State.ERROR
```

Comment:
```suggestion
            type_ = charge.attrib["type"].lower()
            match(type_):
                case "slotcharging" | "slot_charging" | "wirecharging":
                    status = State.DOCKED
                case "idle":
                    status = State.IDLE
                case "going":
                    status = State.RETURNING
                case _:
                    status = State.ERROR
```

Reply:
@edenhaus thanks for your helpful feedback. As suggested, I will make one PR per command and I will add tests.

I have changed this PR to GetChargeState, since I have to understand how to deal with the angle and pos of the GetPos command.

When this PR is fine, I will continue with other commands. Feedback is welcome! :)